src/lsdb/io/to_hipscat.py

```diff
@@ -0,0 +1,149 @@
+    base_catalog_path: str,
+    catalog_name: Union[str, None] = None,
+    storage_options: Union[Dict[Any, Any], None] = None,
+    **kwargs,
+):
+    """Writes a catalog to disk, in HiPSCat format. The output catalog comprises
+    partition parquet files and respective metadata, as well as JSON files detailing
+    partition, catalog and provenance info.
+
+    Args:
+        catalog (Catalog): A catalog to export
+        base_catalog_path (str): Location where catalog is saved to
+        catalog_name (str): The name of the output catalog
+        storage_options (dict): Dictionary that contains abstract filesystem credentials
+        **kwargs: Arguments to pass to the parquet write operations
+    """
+    # Create base directory
+    base_catalog_dir_fp = hc.io.get_file_pointer_from_path(base_catalog_path)
+    hc.io.file_io.make_directory(base_catalog_dir_fp)
+    # Save partition parquet files
+    pixel_to_partition_size_map = write_partitions(catalog, base_catalog_dir_fp, storage_options, **kwargs)
+    # Save parquet metadata
+    hc.io.write_parquet_metadata(base_catalog_path, storage_options, **kwargs)
+    # Save partition info
+    partition_info = _get_partition_info_dict(pixel_to_partition_size_map)
+    hc.io.write_partition_info(base_catalog_dir_fp, partition_info, storage_options)
+    # Save catalog info
+    new_hc_structure = create_modified_catalog_structure(
+        catalog.hc_structure,
+        base_catalog_path,
+        catalog_name if catalog_name else catalog.hc_structure.catalog_name,
+        total_rows=sum(pi[0] for pi in partition_info.values()),
+    )
+    hc.io.write_catalog_info(
+        catalog_base_dir=base_catalog_path,
+        dataset_info=new_hc_structure.catalog_info,
+        storage_options=storage_options,
+    )
+    # Save provenance info
+    hc.io.write_metadata.write_provenance_info(
+        catalog_base_dir=base_catalog_dir_fp,
+        dataset_info=new_hc_structure.catalog_info,
+        tool_args=_get_provenance_info(new_hc_structure),
+        storage_options=storage_options,
+    )
+
+
+def write_partitions(
+    catalog, base_catalog_dir_fp: FilePointer, storage_options: Union[Dict[Any, Any], None] = None, **kwargs
+) -> Dict[HealpixPixel, int]:
+    """Saves catalog partitions as parquet to disk
+
+    Args:
+        catalog (Catalog): A catalog to export
+        base_catalog_dir_fp (FilePointer): Path to the base directory of the catalog
+        storage_options (dict): Dictionary that contains abstract filesystem credentials
+        **kwargs: Arguments to pass to the parquet write operations
+
+    Returns:
+        A dictionary mapping each HEALPix pixel to the number of data points in it.
+    """
+    pixel_to_partition_size_map = {}
+    for pixel, partition_index in catalog._ddf_pixel_map.items():
+        partition = catalog._ddf.partitions[partition_index].compute()
+        pixel_path = hc.io.paths.pixel_catalog_file(base_catalog_dir_fp, pixel.order, pixel.pixel)
```

Comment:
You can fetch the pixel directory path here with `pixel_directory`, instead of parsing it out of the full parquet file path later (https://github.com/astronomy-commons/hipscat/blob/main/src/hipscat/io/paths.py#L22).

This might be more straightforward as:

```
for pixel:
  io.make_directory(io.pixel_dir(base_dir, pixel))
  io.write_dataframe_to_parquet(...)
```

You wouldn't need the new `write_dataframe_to_parquet` method in this case.

Reply:
I agree, thanks for the suggestion!